downstairs/src/lib.rs

```diff
@@ -3472,17 +3260,11 @@ impl Work {
      * return the job ID and the upstairs UUID, moving the state of the job as
      * InProgress. If the dependencies are not met, move the state to DepWait.
      *
-     * If this job is not new, then just return none. This can be okay as we
-     * build our work list with the new_work fn above, but we drop and re-aquire
-     * the Work mutex and things can change.
+     * If this job is not new, then just return None.
      *
      * If the job is InProgress, return itself.
      */
-    fn in_progress(
-        &mut self,
-        ds_id: JobId,
-        log: Logger,
-    ) -> Option<(JobId, UpstairsConnection)> {
```

Comment:
The comment below here, I'm not sure if that condition still exists.  This task now owns the work,
so we can't have any other task (nor ever will) changing the state of things, right?

Reply:
Good eye!  This is fixed in the next PR [here](https://github.com/oxidecomputer/crucible/pull/1341/files#diff-ccb2591a32a7a66c4c4cbb1125ce4e72d2d1b2f10e805991a30f9e8643e3d954L3274)